lua/wire/zvm/zvm_opcodes.lua

```diff
@@ -1231,128 +1231,181 @@ end
 
 --------------------------------------------------------------------------------
 ZVM.OpcodeTable[250] = function(self)  --VADD
+  local seg1code,seg2code = "0","0"
+  if not self.EmitOperandSegment[1] then
+    seg1code = "VM.DS"
+  end
+  if not self.EmitOperandSegment[2] then
+    seg2code = "VM.DS"
+  end
```

Comment:
Shouldn't seg1code and seg2code be `self.EmitOperandSegment[1]` and `self.EmitOperandSegment[2]` if they are set?

Reply:
EmitOperandSegment needs to be checked for validity if that's the case otherwise it still poses a threat.

Reply:
This is for fixing #26, which is that if they have a segment set it would add the segment to it twice, because if say, FS = 6 and R0 = 2, passing FS:R0 to these instructions is the same as passing 8, but these instructions expect it to just be the value of R0 rather than the value of FS + R0, so it would end up trying to load a vector from memory index 14(6+8) rather than index 8(6+2)

The only time this isn't the case is when a segment isn't used, where it instead defaults to adding VM.DS, because the value of the operand isn't getting segment added to it by default, so I default it to adding 0 if there's a segment in that spot to avoid adding it twice without breaking the functionality of non-memory uses of segment:reg offsetting, because reading variables in functions use this trick(they do RSTACK REG,EBP:const a lot for 3 byte access that doesn't modify the ptr register)